[#55190] Fix Tab Nav vertical spacing in Page Header

dominic-braeunlein opened the pull request:
https://community.openproject.org/work_packages/55190

### What are you trying to accomplish?
 Fix Tab Nav vertical spacing in Page Header when deployed

### Screenshots
Before:
![image](https://github.com/opf/primer_view_components/assets/1113942/375dbf56-acd6-4cf1-a482-f62a0a2ec6f4)

After:
<img width="1728" alt="Screenshot 2024-05-28 at 19 13 26" src="https://github.com/opf/primer_view_components/assets/1113942/3d1a9347-7084-4c2b-b3e6-b4f4ef24a0d9">

#### Risk Assessment

- [x] **Low risk** the change is small, highly observable, and easily rolled back.
- [ ] **Medium risk** changes that are isolated, reduced in scope or could impact few users. The change will not impact library availability.
- [ ] **High risk** changes are those that could impact customers and SLOs, low or no test coverage, low observability, or slow to rollback.

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews (Lookbook)
- [x] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

HDinger reviewed (commented):
That feels very very weird.  Could you briefly explain where the actual issue comes from? If the issue only appears when using the component in the core, we should rather investigate there and remove the interference instead of changing the component.

HDinger commented on app/components/primer/open_project/page_header.pcss:
```diff
@@ -66,7 +66,7 @@
-.PageHeader-tabNav {
+.PageHeader-tabNav, .PageHeader-tabNav.tabnav {
   margin-top: var(--stack-gap-normal);
   margin-bottom: 0;
```
Why are we overriding the margin bottom here? It makes total sense to have a margin to the bottom and it is the default behaviour of the tabs..

dominic-braeunlein replied:
The reason I originally removed margin bottom from .tabnav was the 2x1rem margin-bottom in the bottom; From the PageHeader and from the TabNav.

HDinger reviewed (changes requested):
Instead of overriding the tabNav styles, I'd rather add a margin bottom to the elements that require it. Thus we can also be sure, that the spacing is the same. 

So my suggestion would be to remove the styles for `.PageHeader-tabNav` and either add the spacing to the description (the title already has a margin bottom):
```
.PageHeader-description {
   ...
   margin-bottom: var(--stack-gap-normal);
 }  
```

or remove all custom bottom spacings from the single classes and replace it with a `row-gap`

```
.PageHeader {
   ... 
   row-gap: XYrem;
}

dominic-braeunlein commented:
Hi @HDinger, I pushed a new update and wrote some info about my implementation choices

- The reason I originally removed margin bottom from `.tabnav` was the 2x1rem `margin-bottom` in the end; From the PageHeader and from the TabNav. Now I keep the original margin bottom in `.tabnav` and removed it from the `.PageHeader` when TabNav is rendered. 
- I also renamed the modifier `--noBorder` to `--withTabNav` to give it more context as it does more than just removing the bottom border now.
- I could not use `gap` as the vertical spacing of the TabNav is larger than the other spacings. To solve that I used the `.PageHeader--withTabNav` for the `.PageHeader-description`; There is only a `margin-bottom` on the description when TabNav renders. 
```
.PageHeader--withTabNav .PageHeader-description {
  margin-bottom: var(--space-xlarge);
}
```

HDinger reviewed (approved):
Lgtm 👍 Thanks for the extra round, the css looks much better now 🙇
